examples/docker/MLproject

```diff
@@ -3,6 +3,16 @@ name: docker-example
 docker_env:
   image:  mlflow-docker-example
 
+kubernetes_env:
```

Comment:
Mentioned this briefly above but going to repeat just in case. We shouldn't put things like `job_namespace`, `image_namespace` and `registry` or other such arguments here which are not generalizable to all users of this MLProject.

Reply:
I see, so right now we would have just the resources stuff, related to pods quotas and limits. I think that we could then, remove the `kubernetes_env` section and let everything in the backend_config.
Do you agree?

Reply:
Hi @andrewmchen ! 

I finished all the fixes and improvements in order to use this feature with the refactoring that @sueann did.